ts/indicators/keltner-channels.src.ts

```diff
@@ -0,0 +1,258 @@
+    EMA = H.seriesTypes.ema,
+    ATR = H.seriesTypes.atr,
+    merge = H.merge,
+    correctFloat = H.correctFloat;
+
+/**
+ * The Keltner Channels series type.
+ *
+ * @private
+ * @class
+ * @name Highcharts.seriesTypes.keltnerchannels
+ *
+ * @augments Highcharts.Series
+ */
+H.seriesType<Highcharts.KeltnerChannelsIndicator>(
+    'keltnerchannels',
+    'sma',
+    /**
+     * Keltner Channels. This series requires the `linkedTo` option to be set
+     * and should be loaded after the `stock/indicators/indicators.js`,
+     * `stock/indicators/atr.js`, and `stock/ema/.js`.
+     *
+     * @sample {highstock} stock/indicators/keltner-channels
+     *         Keltner Channels
+     *
+     * @extends      plotOptions.sma
+     * @since        7.0.0
+     * @product      highstock
+     * @excluding    allAreas, colorAxis, compare, compareBase, joinBy, keys,
+     *               navigatorOptions, pointInterval, pointIntervalUnit,
+     *               pointPlacement, pointRange, pointStart,showInNavigator,
+     *               stacking
+     * @optionparent plotOptions.keltnerchannels
+     */
+    {
+        params: {
+            period: 20,
+            /**
+             * The ATR period.
+             */
+            periodATR: 10,
+            /**
+             * The ATR multiplier.
+             */
+            multiplierATR: 2
+        },
+        /**
+         * Bottom line options.
+         *
+         */
+        bottomLine: {
+            /**
+             * Styles for a bottom line.
+             *
+             */
+            styles: {
+                /**
+                 * Pixel width of the line.
+                 */
+                lineWidth: 1,
+                /**
+                 * Color of the line. If not set, it's inherited from
+                 * `plotOptions.keltnerchannels.color`
+                 */
+                lineColor: undefined
+            }
+        },
+        /**
+         * Top line options.
+         *
+         * @extends plotOptions.keltnerchannels.bottomLine
+         */
+        topLine: {
+            styles: {
+                lineWidth: 1,
+                lineColor: undefined
+            }
+        },
+        tooltip: {
+            pointFormat: '<span style="color:{point.color}">\u25CF</span><b> {series.name}</b><br/>Upper Channel: {point.top}<br/>EMA({series.options.params.period}): {point.middle}<br/>Lower Channel: {point.bottom}<br/>'
+        },
+        marker: {
+            enabled: false
+        },
+        dataGrouping: {
+            approximation: 'averages'
+        },
+        lineWidth: 1
+    },
+    /**
+     * @lends Highcharts.Series#
+     */
+    merge(multipleLinesMixin, {
+        pointArrayMap: ['top', 'middle', 'bottom'],
+        pointValKey: 'middle',
+        nameBase: 'Keltner Channels',
+        nameComponents: ['period', 'periodATR', 'multiplierATR'],
+        linesApiNames: ['topLine', 'bottomLine'],
+        requiredIndicators: ['ema', 'atr'],
```

Comment:
All these indicator properties are missing in the namespace declaration of the indicator. Additionally you should cast this second merge parameter to the indicator type.

Reply:
Thanks missed it. I was a bit too quick to submit this PR. As for the optional properties in the implemented interface I understand it is needed, but do we also explicitly add them again even when they are present in the parent class?

Reply:
Ok, we should explicitly add the properties even though they are declared in parent class? A good practice, although not strictly needed?

Reply:
Correct. The general problem is, that we are still in the transition phase and because of this, types might change or properties get optional. Conflicts can be easier detected, if every declaration contains the properties that are set during initial run.